lib/telephony/pinpoint/voice_sender.rb

```diff
@@ -45,6 +67,11 @@ def handle_pinpoint_error(err)
         )
       end
 
+      def notify_pinpoint_failover(error)
+        # TODO: log some sort of message?
+        Telephony.config.logger.warn "error region: region"
```

Comment:
cc @pauldoomgov can you help me figure out what kind of warning, maybe even what format, we should log a warning here when we fail over? This will end up in `telephony.log`

Reply:
Thus far it seems we log one JSON line per-SMS.  We don't have a region defined.  So first I would say extend the normal message:
~~~
{
    "success": true,
    "errors": {},
    "request_id": "61526818-bd93-476b-bb6b-6c34efaf9655",
    "delivery_status": "SUCCESSFUL",
    "message_id": "s0f82mp1v3c9ultqqvjpv0gc6t7f4khsf4vqat80",
    "status_code": 200,
    "status_message": "MessageId: s0f82mp1v3c9ultqqvjpv0gc6t7f4khsf4vqat80",
    "adapter": "pinpoint",
    "region": "us-west-2",
    "channel": "sms",
    "context": "authentication"
}
~~~

And then two variations.  The first would be if a failover occurred, but the message was delivered:
~~~
 {
    "success": true,
    "errors": {
        "region_failover": {
            "status_code": "500",
            "region": "us-west-1",
            "status_message": "Uhhhh.... SMS don't feel so goood...."
        }
    },
    "request_id": "61526818-bd93-476b-bb6b-6c34efaf9655",
    "delivery_status": "SUCCESSFUL",
    "message_id": "s0f82mp1v3c9ultqqvjpv0gc6t7f4khsf4vqat80",
    "status_code": 200,
    "status_message": "MessageId: s0f82mp1v3c9ultqqvjpv0gc6t7f4khsf4vqat80",
    "adapter": "pinpoint",
    "region": "us-east-1",
    "channel": "sms",
    "context": "authentication"
}
~~~

And another if both failed:
~~~
 {
    "success": false,
    "errors": {
        "region_failover": {
            "status_code": "500",
            "region": "us-west-1",
            "status_message": "Uhhhh.... SMS don't feel so goood...."
        },
       "other_stuff_that_normally_goes_here": "yep"
    },
    "request_id": "61526818-bd93-476b-bb6b-6c34efaf9655",
    "delivery_status": "FAILURE",
    "message_id": "s0f82mp1v3c9ultqqvjpv0gc6t7f4khsf4vqat80",
    "status_code": 500,
    "status_message": "Dunno...  Maybe try sending a telegram?",
    "adapter": "pinpoint",
    "region": "us-east-1",
    "channel": "sms",
    "context": "authentication"
}
~~~

This should prevent double-counting message failures or (maybe?) breaking any existing parsing.

Reply:
We already log the message delivery status, so this is a separate message, if we want one.